diff --git a/node_modules/@react-native/gradle-plugin/build.gradle.kts b/node_modules/@react-native/gradle-plugin/build.gradle.kts
index 80021f0..f7943d1 100644
--- a/node_modules/@react-native/gradle-plugin/build.gradle.kts
+++ b/node_modules/@react-native/gradle-plugin/build.gradle.kts
@@ -5,47 +5,67 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+import org.gradle.api.internal.classpath.ModuleRegistry
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
 plugins {
-  alias(libs.plugins.kotlin.jvm).apply(false)
-  alias(libs.plugins.ktfmt).apply(true)
+  kotlin("jvm") version "1.7.22"
+  id("java-gradle-plugin")
 }
 
-tasks.register("build") {
-  dependsOn(
-      ":react-native-gradle-plugin:build",
-      ":settings-plugin:build",
-      ":shared-testutil:build",
-      ":shared:build",
-  )
+repositories {
+  google()
+  mavenCentral()
 }
 
-tasks.register("clean") {
-  dependsOn(
-      ":react-native-gradle-plugin:clean",
-      ":settings-plugin:clean",
-      ":shared-testutil:clean",
-      ":shared:clean",
-  )
+gradlePlugin {
+  plugins {
+    create("react") {
+      id = "com.facebook.react"
+      implementationClass = "com.facebook.react.ReactPlugin"
+    }
+  }
 }
 
-tasks.named("ktfmtCheck") {
-  dependsOn(
-      ":react-native-gradle-plugin:ktfmtCheck",
-      ":settings-plugin:ktfmtCheck",
-      ":shared-testutil:ktfmtCheck",
-      ":shared:ktfmtCheck",
-  )
+group = "com.facebook.react"
+
+dependencies {
+  implementation(gradleApi())
+  implementation("com.android.tools.build:gradle:7.4.2")
+  implementation("com.google.code.gson:gson:2.8.9")
+  implementation("com.google.guava:guava:31.0.1-jre")
+  implementation("com.squareup:javapoet:1.13.0")
+
+  testImplementation("junit:junit:4.13.2")
+
+  testRuntimeOnly(
+      files(
+          project.objects.newInstance(ModuleRegistry::class.java)
+              .getModule("gradle-tooling-api-builders")
+              .classpath
+              .asFiles
+              .first()))
 }
 
-tasks.named("ktfmtFormat") {
-  dependsOn(
-      ":react-native-gradle-plugin:ktfmtFormat",
-      ":settings-plugin:ktfmtFormat",
-      ":shared-testutil:ktfmtFormat",
-      ":shared:ktfmtFormat",
-  )
+java {
+  sourceCompatibility = JavaVersion.VERSION_11
+  targetCompatibility = JavaVersion.VERSION_11
 }
 
-// We intentionally disable the `ktfmtCheck` tasks as the formatting is primarly handled inside
-// fbsource
-allprojects { tasks.withType<com.ncorti.ktfmt.gradle.tasks.KtfmtCheckTask>() { enabled = false } }
+tasks.withType<KotlinCompile> {
+  kotlinOptions {
+    jvmTarget = JavaVersion.VERSION_11.majorVersion
+    apiVersion = "1.5"
+    languageVersion = "1.5"
+  }
+}
+
+tasks.withType<Test>().configureEach {
+  testLogging {
+    exceptionFormat = TestExceptionFormat.FULL
+    showExceptions = true
+    showCauses = true
+    showStackTraces = true
+  }
+}
